functions/src/modules/mewm/events/listEvents.ts

```diff
@@ -1,5 +1,38 @@
-import { ExpressFunction } from "../../../@types";
+import type { ExpressFunction } from "../../../@types";
+import { firebaseApp } from "../../../config/firebaseConfig";
+import { expressErrorHandlerFactory } from "../../../helpers";
+import { uasPermissionSwitch } from "../../../systems/uas";
 
-const replaceMe: ExpressFunction = (req, res, next) => {};
+const listEvents: ExpressFunction = (req, res, next) => {
+  uasPermissionSwitch({
+    organizer: { accepted: execute },
+    sponsor: { accepted: execute },
+    mentor: { accepted: execute },
+    volunteer: { accepted: execute },
+    hacker: { accepted: execute },
+  })(req, res, next);
+};
 
-export default replaceMe;
+const execute: ExpressFunction = (req, res, next) => {
+  const errorHandler = expressErrorHandlerFactory(req, res, next);
+
+  firebaseApp
+    .firestore()
+    .collection("miniEvents")
+    .orderBy("eventName", "asc")
+    .get()
+    .then((documents) => {
+      const miniEvents: METMMiniEvent[] = [];
+      for (const doc of documents.docs) {
+        miniEvents.push(doc.data() as METMMiniEvent);
```

Comment:
Heads up! It's the "Mini Events and Workshops Module" (MEWM). Also, I think that these are just referred to as "Events" and not "Mini Events" as they can also include things like deadlines or meals.
```suggestion
        events.push(doc.data() as MEWMEvent);
```